server/chat-commands/admin.ts

```diff
@@ -602,16 +608,57 @@ export const commands: ChatCommands = {
 			this.checkCan('lockdown');
 		}
 
-		let buf = `<strong>${process.pid}</strong> - Main<br />`;
+		const processes = new Map<string, ProcessData>();
+
+		const psOutput = child_process.execSync('ps -o pid,%cpu,time,command', {cwd: `${__dirname}/../..`}).toString();
+		const rows = psOutput.split('\n').slice(1); // first line is the table header
+		for (const row of rows) {
+			if (!row.trim()) continue;
+			const [pid, cpu, time, ...rest] = row.split(' ').filter(Boolean);
+			const entry: ProcessData = {cmd: rest.join(' ')};
+			if (time && time !== '00:00:00') entry.time = time;
+			if (cpu && cpu !== '0.0') entry.cpu = `${cpu}%`;
+			processes.set(pid, entry);
+		}
+
+		let buf = `<strong>${process.pid}</strong> - Main `;
+		const mainProcess = processes.get(`${process.pid}`)!;
+		if (mainProcess.cpu) buf += `(CPU ${mainProcess.cpu}`;
+		if (mainProcess.time) buf += mainProcess.cpu ? `, time: ${mainProcess.time})` : `(time: ${mainProcess.time})`;
+		buf += `<br /><br /><strong>Process managers:</strong><br />`;
+		processes.delete(`${process.pid}`);
+
 		for (const manager of ProcessManager.processManagers) {
 			for (const [i, process] of manager.processes.entries()) {
-				buf += `<strong>${process.getProcess().pid}</strong> - ${manager.basename} ${i} (load ${process.load})<br />`;
+				const pid = process.getProcess().pid;
+				buf += `<strong>${pid}</strong> - ${manager.basename} ${i} (load ${process.load}`;
+				const info = processes.get(`${pid}`)!;
+				if (info.cpu) buf += `, CPU: ${info.cpu}`;
+				if (info.time) buf += `, time: ${info.time}`;
+				buf += `)<br />`;
+				processes.delete(`${pid}`);
 			}
 			for (const [i, process] of manager.releasingProcesses.entries()) {
-				buf += `<strong>${process.getProcess().pid}</strong> - PENDING RELEASE ${manager.basename} ${i} (load ${process.load})<br />`;
+				const pid = process.getProcess().pid;
+				buf += `<strong>${pid}</strong> - PENDING RELEASE ${manager.basename} ${i} (load ${process.load}`;
+				const info = processes.get(`${pid}`)!;
+				if (info.cpu) buf += `, CPU: ${info.cpu}`;
+				if (info.time) buf += `, time: ${info.time}`;
+				buf += `)<br />`;
+				processes.delete(`${pid}`);
 			}
 		}
-
+		buf += `<br />`;
+		buf += `<strong>Other processes:</strong><br />`;
+
+		for (const [pid, process] of processes) {
+			buf += `<strong>${pid}</strong> - <code>${process.cmd}</code>`;
+			if (process.cpu) buf += ` (CPU: ${process.cpu}`;
+			if (process.time) {
+				buf += `${process.cpu ? `, ` : ' ('}time: ${process.time})`;
+			}
+			buf += `<br />`;
+		}
```

Comment:
This should be in a readmore; it's too spammy to show by default.

Reply:
Done.

Reply:
It plausibly shouldn't be shown at all, but I guess we can try it out for now and see how it goes.